nsapass

```diff
@@ -166,8 +167,16 @@ def load():
 def save(d):
     print_info('saving db...')
     data = json.dumps(d, indent=4)
-    cmd(ENCCMD, stdin=data.encode())
-    # TODO: replace the file atomically -- unless scrypt does that for us?
+    tempDB = tempfile.NamedTemporaryFile(prefix=DB, delete=False)
+    delete = True
+    try:
+        cmd([p.format(tempDB.name) for p in ENCCMD], stdin=data.encode())
+        os.rename(tempDB.name, DB)
+        delete = False
+    finally:
+        if delete:
+            os.unlink(tempDB.name)
+        tempDB.close()
```

Comment:
Note that I gave up on using the internal cleanup that `tempfile` provides. The previous version looked like this,
```python
    with tempfile.NamedTemporaryFile(prefix=DB) as tempDB:
         cmd([p.format(tempDB.name) for p in ENCCMD], stdin=data.encode()) 
         os.rename(tempDB.name, DB) 
         os.link(DB, tempDB.name) # closing tempDB fails noisily without this
```
which is more concise and may look cleaner, but suffers from a potential (though very unlikely) race between the `os.link` and other programs creating a file with that name.

In a better world, `tempfile` objects would provide a way of disabling the cleanup.

Reply:
first off, i must say, this is a very _glorious_ pr at so many levels:

- level 0:  it's pr number _2_, which is the smallest prime number that is also an even number.
- level 1: a seemingly simple problem spiralled into a major ocd case; often happens with prime numbers!

i totally agree with everything you said, and your solution seems perfect to me. in fact, i've just spent an ocd session worth of `~100` calories in it, and i doubt that anyone can come up with anything better (anyone dares?)

going to merge soon..

thanks a lot! what a legendary pr-er!